### PR TITLE
Multi calendar view

### PR DESCRIPTION
**Pull request by steppy452**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
* adds ability to specify number of months to show
* adds ability to turn on day of week labels
* adds ability to customize day of week labels, but first looks at user's browser settings
* adds color tokens for the customizing look and feel of calendar
* fixes some minor stying bits
* fixes a bug where value can be undefined, but display shows "today" as being selected/highlighted by default
* fixes rounding of corners !!!! 😮‍💨 

image

image

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by amcdnl — COMMENTED** [review]
Interesting - I would have tried to make a new calendar called like `CalendarRange` or something and then consume the child containers and add logic around them. This is probably fine but what do you think about feasibility of that?

**Comment by amcdnl** [comment]
> oh yea, that's definitely feasible. i can make that update

Awesome. That will make it more modular.

**Comment by steppy452** [comment]
@amcdnl actually, I think I take that back - if both calendar's aren't rendered in the same component, then it won't highlight correctly because the hovering date is handled internally on the CalendarDays component
https://github.com/reaviz/reablocks/blob/master/src/form/Calendar/CalendarDays/CalendarDays.tsx#L88

I forgot I tried this before and ended up with the highlighting issue

File: src/form/Calendar/CalendarDays/CalendarDays.tsx [hunk 1]
```diff
@@ -131,12 +171,36 @@ export const CalendarDays: FC<CalendarDaysProps> = ({
         Array.isArray(current) &&
         isSameDay(addDays(nextDayRangeEnd, -1), day.date);
 
+      // Determine styling of range start and end dates
```
Comment (steppy452) [inline src/form/Calendar/CalendarDays/CalendarDays.tsx:174]: this is a little messy, but trying to determine whether or not to show the rounded corners wasn't as straightforward as you'd think

File: src/form/Calendar/CalendarDays/CalendarDays.tsx [hunk 2]
```diff
@@ -74,50 +100,64 @@ const ZERO_TIME = {
   milliseconds: 0
 };
 
+const DAY_OF_WEEK_LABELS = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
```
Comment (steppy452) [inline src/form/Calendar/CalendarDays/CalendarDays.tsx:103]: maybe need to allow for customization in the future?

File: src/form/Calendar/Calendar.story.tsx [hunk 3]
```diff
@@ -119,11 +118,13 @@ export const Multiview = () => {
 
   return (
     <Card>
-      <CalendarRange
+      <Calendar
         value={range}
         onChange={val => setRange(val as [Date, Date])}
-        numMonths={3}
-        enableDayOfWeek
+        numMonths={2}
```
Comment (amcdnl) [inline src/form/Calendar/Calendar.story.tsx:136]: @steppy452 - I imagined `CalendarRange` to be the top most component, not the calendar.

**Review by amcdnl — CHANGES_REQUESTED** [review]

File: src/form/Calendar/Calendar.story.tsx (hunk 3 above)
Comment (steppy452) [reply src/form/Calendar/Calendar.story.tsx:136]: so I tried that - it basically just duplicated the calendar code and added the same logic .. unless you don't think we need the ability to switch into month/year views

**Review by amcdnl — CHANGES_REQUESTED** [review]

File: src/form/Calendar/CalendarDays/CalendarDays.tsx (hunk 2 above)
Comment (amcdnl) [reply src/form/Calendar/CalendarDays/CalendarDays.tsx:103]: Something like this - https://ionel-lupu.com/blog/how-to-display-weekday-names-in-your-users-language

File: src/form/Calendar/CalendarDays/CalendarDays.tsx [hunk 4]
```diff
@@ -132,92 +130,103 @@ export const CalendarDays: FC<CalendarDaysProps> = ({
...
+      let isActive = false;
+      let isRangeStart = false;
+      let isRangeEnd = false;
+
+      if (!isRange && isValid(current)) {
+        // if not a range
+        isActive = isSameDay(day.date, current as Date);
+      } else if (!isSelectionStarted) {
+        // if selection has not started
+        isActive = isSameDay(day.date, currentHover);
```
Comment (amcdnl) [inline src/form/Calendar/CalendarDays/CalendarDays.tsx:171]: Can we break this logic out into a pure function w/ a few tests?

**Review by amcdnl — APPROVED** [review]

**Pull request merged**